protected/views/adminDataset/_form.php

```diff
@@ -349,23 +349,27 @@
                                     echo CHtml::ajaxButton(
                                         'Mint DOI',
                                         Yii::app()->createUrl('/adminDataset/mint/'),
-                                        array(
+                                        [
                                             'type' => 'POST',
-                                            'data' => array('doi' => 'js:$("#Dataset_identifier").val()'),
+                                            'data' => ['doi' => 'js:$("#Dataset_identifier").val()'],
                                             'dataType' => 'json',
-                                            'success' => 'js:function(output){
-                                            console.log(output);
-                                            if(output.status){
-                                                $("#minting").removeClass("errorMessage");
-                                                $("#minting").html("new DOI successfully minted");
-
-                                            }else {
-                                                $("#minting").addClass("errorMessage");
-                                                $("#minting").html("error minting a DOI: "+ output.md_curl_status + ", " + output.doi_curl_status);
-                                            }
-                                            $("#mint_doi_button").toggleClass("active");
-                                        }',
-                                        ),
+                                            'success' => new CJavaScriptExpression('function(output) {
+                                                console.log(output);
+                                                if (output.check_metadata_status == 200 && output.check_doi_status == 200 && output.update_md_status == 201) {
+                                                    $("#minting").addClass("alert alert-info").html("This DOI exists in datacite already, no need to mint, but the metadata is updated!");
+                                                } else if (output.check_metadata_status == 200 && output.check_doi_status == 200 && output.update_md_status == 422) {
+                                                    $("#minting").addClass("alert alert-info").html("This DOI exists in datacite, but failed to update metadata because of: " + output.update_md_response);
+                                                } else if (output.create_md_status == 201 && output.create_doi_status == 201) {
+                                                    $("#minting").addClass("alert alert-success").html("new DOI successfully minted");
```

Comment:
It would be good to capitalize the first letter of `new` as this message is shown to user.